### PR TITLE
feat: add "onKeyDown" prop to various input components

### PR DESCRIPTION
**Pull request by Mohammer5**
Relates to [TECH-856](https://dhis2.atlassian.net/browse/TECH-856)

### CODE REVIEW

File: components/file-input/src/file-input-field-with-list/__tests__/file-input-field-with-list.test.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { configure, render, fireEvent, screen } from '@testing-library/react'
+import React from 'react'
+import { FileInputFieldWithList } from '../file-input-field-with-list.js'
+
+configure({ testIdAttribute: 'data-test' })
```
Comment (HendrikThePendric) [inline components/file-input/src/file-input-field-with-list/__tests__/file-input-field-with-list.test.js:5]: I believe this isn't needed? We have configured that already in `./jest.testing-library.config.js`.

**Review by HendrikThePendric — CHANGES_REQUESTED** [review]
I've left one code comment, but it actually applies to most test files. Please remove these lines before merging.

Apart from that, it all looks good to me. Most important thing is that these new event handlers implement the same callback signature as the existing event handlers.

**Review by HendrikThePendric — APPROVED** [review]
Thanks for removing those calls to `configure`

File: components/select/src/single-select-field/single-select-field.js [hunk 2]
```diff
@@ -168,6 +170,7 @@ SingleSelectField.propTypes = {
     onChange: PropTypes.func,
     /** Called with signature `({ selected: string }, event)` */
     onFocus: PropTypes.func,
+    onKeyDown: PropTypes.func,
```
Comment (KaiVandivier) [inline components/select/src/single-select-field/single-select-field.js:173]:
```suggestion
    /** Called with signature `({ selected: string }, event)` */
    onKeyDown: PropTypes.func,
```

File: components/select/src/multi-select-field/multi-select-field.js [hunk 3]
```diff
@@ -169,6 +171,7 @@ MultiSelectField.propTypes = {
     onChange: PropTypes.func,
     /** Called with signature `({ selected: [String] }, event)` */
     onFocus: PropTypes.func,
+    onKeyDown: PropTypes.func,
```
Comment (KaiVandivier) [inline components/select/src/multi-select-field/multi-select-field.js:174]:
```suggestion
    /** Called with signature `({ selected: [String] }, event)` */
    onKeyDown: PropTypes.func,
```

File: components/button/src/button/button.js [hunk 4]
```diff
@@ -151,4 +155,6 @@ Button.propTypes = {
     onClick: PropTypes.func,
     /** Callback to trigger on focus. Called with same args as `onClick` */
     onFocus: PropTypes.func,
+    /** Callback to trigger on focus. Called with same args as `onClick` */
```
Comment (KaiVandivier) [inline components/button/src/button/button.js:158]:
```suggestion
    /** Callback to trigger on key-down. Called with same args as `onClick` */
```

**Review by KaiVandivier — APPROVED** [review]
Just a few prop description edits :) everything else looks good though so will approve!

**Pull request merged**